### ISSUE TITLE
Unhelpful unused_doc_comment warning generated for external macros

### ISSUE THREAD

**Issue by bradfier** [rust-lang/rust#43974]
On Nightly, rustc is generating `warning: doc comment not used by rustdoc` for macro expansions from outside the current crate.

```
warning: doc comment not used by rustdoc
  --> src/main.rs:12:5
   |
12 | /     error_chain!{
13 | |         foreign_links {
14 | |             Serde(::serde_json::Error);
15 | |         }
16 | |     }
   | |_____^
   |
   = note: #[warn(unused_doc_comment)] on by default
   = note: this error originates in a macro outside of the current crate
```

A reproducible example is available at https://github.com/bradfier/external-docstrings-example

Since this behaviour isn't present on Stable, I'm assuming it's unintentional, if it's a new lint, or the semantics of when it is displayed have changed for Nightly, it would be useful if the warning message provided some more information about exactly where the lint is being triggered. At the moment, the produced info isn't much use for finding where the unused doc comment is.

**Comment by oli-obk** [comment]
Such lints need a macro check like clippy has: https://github.com/rust-lang-nursery/rust-clippy/blob/7cdaeae1b877ca03b26ccb9b82754b826b03da5d/clippy_lints/src/utils/mod.rs#L114-L151

**Comment by steveklabnik** [comment]
This has been fixed in newer versions of error-chain, though they have not been released yet https://github.com/rust-lang-nursery/error-chain/issues/208

**Comment by bradfier** [comment]
Thanks @steveklabnik, to be more precise, the fix for this is in [this commit](https://github.com/BenjaminGill-Metaswitch/error-chain/commit/1cf15e6ed364921c3faf0a8587fa5f0072fd02de)

Just need to wait for error-chain 0.11.0 to land!

Would it be possible for rustc to give a printout of the line or file in the macro definition which caused the lint to be triggered?
